### PR TITLE
[BEAM-691] Correctly implement display data for DataflowRunner Read transforms

### PR DESCRIPTION
**Pull request by swegner**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

**Comment by swegner** [comment]
R: @bjchambers

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java [hunk 1]
```diff
@@ -105,7 +105,14 @@ public DataflowUnboundedReadFromBoundedSource(BoundedSource<T> source) {
 
   @Override
   public String getKindString() {
-    return "Read(" + approximateSimpleName(source.getClass()) + ")";
+    String sourceName;
+    if (source.getClass().isAnonymousClass()) {
+      sourceName = "AnonymousSource";
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java]: Why not include the anonymous class name? That would at least tell us which package/class defined it?

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java [hunk 2]
```diff
@@ -105,7 +105,14 @@ public DataflowUnboundedReadFromBoundedSource(BoundedSource<T> source) {
...
+    } else {
+      sourceName = approximateSimpleName(source.getClass());
+    }
+
+    return "Read(" + sourceName + ")";
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java]: I could be wrong, but I think that (1) getKindString acts as the default name of a transform (2) changing names is not update-compatible. If both of these are correct, then changing anonymous sources to be named "Read(AnonymousSource)" would be update-incompatible, which we may want to avoid. Why is this change necessary for this PR?

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSourceTest.java [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Tests for {@see DataflowUnboundedReadFromBoundedSource}.
+ */
+@RunWith(JUnit4.class)
+public class DataflowUnboundedReadFromBoundedSourceTest {
+  /** Source implementation only useful for its identity. */
+  static class NamedSource extends BoundedSource<String> {
```
Comment (bjchambers) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSourceTest.java]: `NoopNamedSource`? Also move to the bottom so the tests are more apparent.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java (hunk 2 above)
Comment (swegner) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java]: `approximateSimpleName(Class<?>)` [fails on anonymous classes](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/util/StringUtils.java#L209), so this change won't affect anything that was previously working. This change matches the [default `PTransform` implementation](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PTransform.java#L242).

I hit the issue during this change because my test was sending an anonymous `Source` class into the transform. I could avoid it by using a concrete class, but no reason to step around an existing bug.

File: runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSourceTest.java (hunk 3 above)
Comment (swegner) [inline runners/google-cloud-dataflow-java/src/test/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSourceTest.java]: Done.

File: runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java (hunk 1 above)
Comment (swegner) [inline runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/internal/DataflowUnboundedReadFromBoundedSource.java]: This follows the convention defined by [`PTransform.getKindString()`](https://github.com/apache/incubator-beam/blob/master/sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/PTransform.java#L242).

**Comment by bjchambers** [comment]
LGTM
